### DISCUSSION TITLE
CMake Generate step keeps failing

### DISCUSSION THREAD

**Discussion by ebenezerArt** [d99kris/nchat#173] [Q&A]
![image](https://github.com/d99kris/nchat/assets/70277659/bbd6a6b8-267f-4ff3-9b1a-b755ae59f460)

How do I set the variable build testing to FOUND and not NOTFOUND.

.....or any other way to get this warning from going away.

**Comment by d99kris** [comment]
Can you please share which command you are running to build/install nchat? 
Also can I check your macOS version (e.g. run command `sw_vers`)?

Copy and pasting the text from your screenshot to facilitate searchability:
```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files: / tmp/nchat-20231222-71749-7j0ppc/nchat-3.67/lib/ncutil/MAGIC_INCLUDE_DIR
used as include directory in directory /tmp/nchat-20231222-71749-7j0ppc/nchat-3.67/lib/ncutil
MAGIC_LIBRARY
linked by target "ncutil" in directory /tmp/nchat-20231222-71749-7j0ppc/nchat-3.67/lib/ncutil
-- Generating done (0.5s)
CMake Warning:
Manually-specified variables were not used by the project:
BUILD_TESTING
CMake Generate step failed. Build files cannot be regenerated correctly.
If reporting this issue please do so at (not Homebrew/brew or Homebrew/homebrew-core) :
https://github.com/d99kris/homebrew-nchat/issues
```

**Reply by ebenezerArt** [reply]
i am using this command:

brew tap d99kris/nchat
brew install nchat

**Reply by ebenezerArt** [reply]
my macOS version is 14.2

**Reply by d99kris** [reply]
Thanks for the quick reply! Can I ask for one more thing, could you please share the full output from 
```brew install nchat```
ideally if you could copy and paste the output here in a comment, or otherwise redirect its output to a text file 
```brew install nchat &> brew.txt```
and attach it in a comment (can just drag an drop into the comment field).

**Reply by ebenezerArt** [reply]
<img width="1141" alt="image" src="https://github.com/d99kris/nchat/assets/70277659/657455e0-e821-483f-b010-1c63e87286c7">

Is this fine

**Reply by d99kris** [reply]
Thanks, that's useful. One more (hopefully last) thing, could you attach a copy of 
```/Users/ebenezerarthur/Library/Logs/Homebrew/nchat/01.cmake```
here. To do so you can open directory in Finder
```open ~/Library/Logs/Homebrew/nchat/```
and just drag `01.cmake` into the comment field here in Github.

**Reply by ebenezerArt** [reply]
i cannot drag it into the comment field it does not support this file type.

can I do something else

**Reply by d99kris** [reply]
Oh I see. Okay you can make a copy of the file, for example:
```cp /Users/ebenezerarthur/Library/Logs/Homebrew/nchat/01.cmake ~/Desktop/01.cmake.txt```
and then upload the copy from the desktop, named `01.cmake.txt`.

**Reply by ebenezerArt** [reply]
[01.cmake.txt](https://github.com/d99kris/nchat/files/13757725/01.cmake.txt)

**Reply by d99kris** [reply]
Thank you 👍  I don't know yet what the actual root cause is, but the installation is failing because `nchat` does not find `libmagic` (which would be automatically installed when installing `nchat` using `brew`). I cannot reproduce the problem on my side (even if uninstalling `libmagic` first). One thing you could try is to uninstall `libmagic`, and then try install `nchat` (in case the `libmagic` installation is corrupted somehow). So you could try:
```
brew uninstall libmagic
brew install nchat
```

**Reply by ebenezerArt** [reply]
thank you i tried this but it still get same error: but it does tell me that the brew link command did not happen successfully

**Reply by ebenezerArt** [reply]
<img width="1138" alt="image" src="https://github.com/d99kris/nchat/assets/70277659/6d351148-69bb-45ed-8024-9526096f97a0">

**Reply by d99kris** [reply]
I'll paste the text here as well, so it's easier to find in the future.
```
lebenezerarthur@2003 ~ % brew install nchat
==> Downloading https://formulae.brew.sh/api/formula.jws.json
################################################################
==> Downloading https://formulae.brew.sh/api/cask.jws.json
##################################################################################
=> Fetching dependencies for d99kris/nchat/nchat: libmagic ==> Fetching libmagic ==>
Downloading https://ghcr.io/v2/homebrew/core/libmagic/manifests/5.45
Already downloaded: /Users/ebenezerarthur/Library/Caches/Homebrew/downloads/7009258a720f4578934a2b2ec9ba7c36fb181470926eb7db67f0f49121becfb--libmagic-5.45. bottle
son
==> Downloading https://ghcr.io/v2/homebrew/core/libmagic/blobs/sha256:41360c07f92f0a4ab86a78048f6c1fa74a0d1192b60ad45954d384d7606adada
Already downloaded: /Users/ebenezerarthur/Library/Caches/Homebrew/downloads/55b719a20662902024980c971b164b4af8ec3f4df6f88da349640f7fc888dd5--libmagic--5.45.sonom
I.gZ
=> Fetching d99kris/nchat/nchat
==> Downloading https://github.com/d99kris/nchat/archive/refs/tags/v3.67.tar.gz
Already downloaded: /Users/ebenezerarthur/Library/Caches/Homebrew/downloads/950ca4671e6b6962e0d83007925c30431516927ad422f17113d2bdecdcfe903--nchat-3.67.tar.gz
==> Installing nchat from d99kris/nchat
ニ＝＞
Installing dependencies for d99kris/nchat/nchat: libmagic
=> Installing d99kris/nchat/nchat dependency: libmagic
==> Downloading https://ghcr.io/v2/homebrew/core/libmagic/manifests/5.45
Already downloaded: /Users/ebenezerarthur/Library/Caches/Homebrew/downloads/7009258a720f4578934a2b2ec9ba7c36fb181470926eb7db67f0f49121becfb--libmagic-5.45.bottle
son
=> Pouring
libmagic--5.45.sonoma.bottle.tar.gz
Error: The brew link step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink share/misc/magic
/usr/local/share/misc is not writable.
You can try again using:
brew link libmagic
Summary
/usr/local/Cellar/libmagic/5.45: 356 files, 10. 2MB
Installing d99kris/nchat/nchat cmake .•
Last 15 lines from /Users/ebenezerarthur/Library/Logs/Homebrew/nchat/01. cmake:
CMake Error: The
following variables are used in this project,
but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files: / tmp/nchat-20231222-89594-mgb6fg/nchat-3.67/lib/ncutil/MAGIC_INCLUDE_DIR
used as
include directory in directory /tmp/nchat-20231222-89594-mgb6fg/nchat-3.67/lib/ncutil
MAGIC_LIBRARY
linked by target "ncutil" in directory / tmp/nchat-20231222-89594-mgb6fg/nchat-3.67/lib/ncutil
-- Generating done (0.5s)
CMake Warning:
Manually-specified variables were not used by the project:
BUILD_TESTING
CMake Generate step failed. Build files cannot be regenerated correctly.
If
reporting this issue please do so at (not Homebrew/brew or Homebrew/homebrew-core):
https://github.com/d99kris/homebrew-nchat/issues
```

**Reply by d99kris** [reply]
I think the relevant part is this:
```
Error: The brew link step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink share/misc/magic
/usr/local/share/misc is not writable.
You can try again using:
brew link libmagic
```
This Stack Overflow [answer](https://stackoverflow.com/a/14539521/485047) suggests to change ownership of brew install directories. I think it can be worth a shot. I.e. try:
```
sudo chown -R $(whoami) $(brew --prefix)/*
```
Then try install/link libmagic:
```
brew link libmagic
```
If it works without errors/warnings, then you can proceed to install nchat.

If it has errors, perhaps you can share the output of
```
ls -l /usr/local/share
```
and maybe also
```
ls -l /usr/local/share/misc
```
Thanks!

**Reply by ebenezerArt** [reply]
thanks, changing ownership of the brew install worked. the install went through nicely.

**Reply by d99kris** [reply]
Great, thanks for sharing 👍
